src/Altinn.AccessManagement.Core/Services/PolicyAdministrationPoint.cs

```diff
@@ -56,6 +58,168 @@ public async Task<bool> WritePolicyAsync(string org, string app, Stream fileStre
         }
 
+        private async Task<bool> WriteInstanceDelegationPolicyInternal(string policyPath, InstanceRight rules, CancellationToken cancellationToken = default)
+        {
+            var policyClient = _policyFactory.Create(policyPath);
+            if (!await policyClient.PolicyExistsAsync(cancellationToken))
+            {
+                // Create a new empty blob for lease locking
+                await policyClient.WritePolicyAsync(new MemoryStream(), cancellationToken: cancellationToken);
+            }
+
+            string leaseId = await policyClient.TryAcquireBlobLease(cancellationToken);
+            if (leaseId != null)
+            {
+                try
+                {
+                    // Check for a current delegation change from postgresql
+                    InstanceDelegationChangeRequest requestLastChanged = new InstanceDelegationChangeRequest
+                    {
+                        FromType = rules.FromType,
+                        FromUuid = rules.FromUuid,
+                        ToType = rules.ToType,
+                        ToUuid = rules.ToUuid,
+                        Resource = rules.ResourceId,
+                        InstanceDelegationMode = rules.InstanceDelegationMode,
+                        Instance = rules.Instance
+                    };
+
+                    InstanceDelegationChange currentChange =
+                        await _delegationRepository.GetLastInstanceDelegationChange(requestLastChanged,
+                            cancellationToken);
+
+                    XacmlPolicy existingDelegationPolicy = null;
+                    if (currentChange != null && currentChange.DelegationChangeType != DelegationChangeType.RevokeLast)
+                    {
+                        policyPath = currentChange.BlobStoragePolicyPath;
+                        existingDelegationPolicy = await _prp.GetPolicyVersionAsync(policyPath,
+                            currentChange.BlobStorageVersionId, cancellationToken);
+                    }
+
+                    // Build delegation XacmlPolicy either as a new policy or add rules to existing
+                    XacmlPolicy delegationPolicy;
+                    if (existingDelegationPolicy != null)
+                    {
+                        delegationPolicy = existingDelegationPolicy;
+                        PolicyHelper.GetPolicyDataFromInstanceRight(rules, out string resourceId, out string fromType, out string fromId, out string toType, out string toId, out string performedById, out string performedByType);
+
+                        foreach (InstanceRule rule in rules.InstanceRules)
+                        {
+                            if (!DelegationHelper.PolicyContainsMatchingInstanceRule(delegationPolicy, rule))
+                            {
+                                // TODO: Decide how From and To should be stored to policy file.
```

Comment:
Ang. ToDo:
From lagrer vi vel ikke i policy?
To so bør vi vel nå bruke uuid'ene, selv om dette betyr at vi må inn med en del ekstra enrichment i PDP for å kunne gjøre policy matchingen.

Reply:
Fjernet Todo og vi lagrer From bare som komentar slik at det kan leses hva en regel gjelder uten å lese hele regelen.